projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -20,23 +20,33 @@ import {
 import { ClrDatagrid, ClrDatagridPagination, ClrDatagridStateInterface } from '@clr/angular';
 import { LazyString, TranslationService } from '@vcd/i18n';
 import { Observable } from 'rxjs';
+import { ActionMenuComponent, DEFAULT_ACTION_DISPLAY_CONFIG } from '../action-menu/action-menu.component';
 import { ActivityReporter } from '../common/activity-reporter';
-import { TextIcon } from '../common/interfaces/action-item.interface';
+import {
+    ActionDisplayConfig,
+    ActionHandlerType,
+    ActionItem,
+    ActionStyling,
+    ActionType,
+} from '../common/interfaces/action-item.interface';
 import { SubscriptionTracker } from '../common/subscription';
 import { TooltipSize } from '../lib/directives/show-clipped-text.directive';
 import { DatagridFilter } from './filters/datagrid-filter';
 import { ComponentRendererConstructor, ComponentRendererSpec } from './interfaces/component-renderer.interface';
 import {
-    Button,
-    ButtonConfig,
     ColumnRendererSpec,
-    ContextualButtonPosition,
     FunctionRenderer,
     GridColumn,
     GridColumnHideable,
-    InactiveButtonDisplayMode,
 } from './interfaces/datagrid-column.interface';
-import { ContextualButton } from './interfaces/datagrid-column.interface';
+
+/**
+ * An enum that describes where the contextual buttons should display.
+ */
+export enum ContextualActionPosition {
```

Comment:
Card view doesn't have to worry about this. Since action menu is part of our common datagrid component, the caller has to tell the datagrid where to display the action menu component.

In the case of card, the actions are always inside the card. If they have to be displayed on the top, that is the responsibility of card container component.